packages/mds-types/utils.ts

```diff
@@ -187,13 +187,6 @@ export type Json = Nullable<JsonValue>
 export type AnyFunction<A = any> = (...args: any[]) => A
 export type AnyConstructor<A = object> = new (...args: any[]) => A
 
-/**
- * Intersection of T and U
- * @example
- * type A = SetIntersection<'a' | 'b', 'a' | 'c'> // 'a'
- */
-export type SetIntersection<T extends keyof any, U extends keyof any> = T extends U ? T : never
```

Comment:
This type was totally unnecessary, and it could be derived by just using the `&` symbol to get the intersection between two literal types. So instead of `SetIntersection<'a' | 'b', 'b' | 'c'> // returning 'b'` you could literally just do `('a' | 'b') & ('b' | 'c')` d'oh.